app.py

```diff
@@ -184,28 +176,102 @@ def main():
     pull_cache_from_hub = arguments.pull_cache_from_hub
 
     # Initialize the interface and grab the UI-provided arguments
```

Comment:
I prefer to have the `main` function serve as a basic entry point for the program, calling to other functions. Each function should be modularized so that each one does one task, rather than having all the code together in one long function. I'm not sure whether this is a kind of `official` practice, but it's definitely what I'm used to and what I'm used to seeing. It helps with debugging and reusing functionality.

Reply:
Good point - most of the gradio demo creation got delegated to `create_demo` so main just parses the command line args and calls `create_demo` now